proto/account.proto

```diff
@@ -0,0 +1,50 @@
+syntax = "proto3";
+
+import "trees.proto";
+
+// The account type.
+//
+// Every account has a designed type that can be inferred from the account id.
+enum AccountType {
+    RegularAccountImmutableCode = 0;
+    RegularAccountUpdatableCode = 1;
+    FungibleFaucet = 2;
+    NonFungibleFaucet = 3;
+}
+
+// A Miden account.
+message AccountId {
+    // A miden account is defined with a little bit of proof-of-work, the id itself
+    // is defined as the first word of a hash digest. For this reason account ids
+    // can be considered as random values, because of that the encoding bellow uses
+    // fixed 64 bits, instead of zig-zag encoding.
+    fixed64 id = 1;
+
+    // This enum exists to decouple the internal representation of the account id
+    // from its type, and to make handling of the account type a bit easier (i.e.
+    // no need for bitmasking).
+    AccountType type = 2;
+}
+
+// A MASM procedure
+message Procedure {
+    int32 num_locals = 1; 
+    bytes code = 2;
+}
+
+message AccountCode {
+    repeated Procedure procedures = 1;
+}
```

Comment:
I think `AccountCode` would be defined more like:
```proto
message AccountCode {
    repeated Digest procedures = 1;
    bytes module = 2;
}
```
This will make it consistent with the [AccountCode](https://github.com/0xPolygonMiden/miden-base/blob/main/objects/src/accounts/code.rs#L22) struct (we don't serialize `procedure_tree` as it can be reconstructed from `procedures`).

But also, I'm not sure we need to get into this level of detail (see next comment).